### PR TITLE
restore ec2 instance from backup script

### PR DESCRIPTION
**Pull request by joshuaroot**
removed iam function & added instance created waiter

### CODE REVIEW

**Comment by kapilt** [comment]
[copying over some of the comments i had from the other pull request, but worth looking over those] so rather the interactive wizard style with screen clearing, i think just two subcommands, one to list snapshots given an instance id, and one to restore with params on the cli would suffice and avoid some odd interaction modes as well as being more portable (ie. work on windows). for the case where the instance no longer exists, i think we need additional cli params re subnet, role, instance type, sec groups, and key for pass through to run_instances

**Comment by joshuaroot** [comment]
When adding in the additional parameters, I am assuming adding additional validation checks for the parameters will be necessary (making sure the role/group/key/etc exists). Does that sound right?

**Comment by kapilt** [comment]
hmmm.. i'd probably let the underlying error from the run instance call surface to the user. up to you. if you let the error go through it would be good to try and pick up the volumes created previously on subsequent runs.

File: tools/ec2recovery.py [hunk 1]
```diff
@@ -1,280 +1,211 @@
...
-    waiter = client.get_waiter('instance_running')
-    waiter.wait(
-        DryRun=dry_run,
-        InstanceIds=[instance_id]
-    )
-    restored.create_tags(
-        DryRun=dry_run,
-        Tags=create_tags(instance['Tags']))
-    return instance_id
+def read_user_date(userdata):
```
Comment (kapilt) [inline tools/ec2recovery.py]: its great that you covered for completeness and more importantly made it optional, but afaics userdata is initialization, and on restore initialization is moot and generally untested and not idempotent, fine with leaving it in, but maybe print a warning if specified..

File: tools/ec2recovery.py [hunk 2]
```diff
@@ -1,280 +1,211 @@
...
+        instances = ec2.create_instances(
+            DryRun=dryrun,
+            ImageId=ami,
+            MinCount=1,
+            MaxCount=1,
+            KeyName=keyname,
+            SecurityGroupIds=securitygroups.split(','),
+            InstanceType=type,
+            SubnetId=subnet,
+            BlockDeviceMappings=snapshots,
```
Comment (kapilt) [inline tools/ec2recovery.py]: we need a little more distinction here, in some cases we are restoring with a partial snapshot set, which i don't think we should fail on, ie. merge snapshots extant against ami block dev, and if perfect union, print a warning.

**Comment by joshuaroot** [comment]
remote branch merge issues. closing pull request & opening new one.

**Comment by kapilt** [comment]
we can do a screen share, i suspect its because your need to merge from upstream master.. ie


what's your git remote show

```$ git remote -v```

mine for example has

```$ git remote -v
$ git remote -v
faisal  https://github.com/fdosani/cloud-custodian.git (fetch)
faisal  https://github.com/fdosani/cloud-custodian.git (push)
joshua  https://github.com/joshuaroot/cloud-custodian.git (push)
joshua  https://github.com/joshuaroot/cloud-custodian.git (fetch)
origin  https://github.com/kapilt/cloud-custodian.git (push)
origin  https://github.com/kapilt/cloud-custodian.git (fetch)
upstream        https://github.com/capitalone/cloud-custodian.git (fetch)
upstream        https://github.com/capitalone/cloud-custodian.git (push)
```

i push my branches to origin for pull requests, and i pull and merge from upstream

**Pull request closed without merging**